lib/errors/MaxRetryExceededError.js

```diff
@@ -1,22 +1,34 @@
-var NAME = "MaxRetryExceededError";
+var applyErrorPrototype = require("./applyErrorPrototype");
+
 /**
  * A request can only be retried up to a specified limit.  Once that
  * limit is exceeded, then an error will be thrown.
  *
+ * @param {*} missingOptimizedPaths
+ *
  * @private
  */
 function MaxRetryExceededError(missingOptimizedPaths) {
-    this.message = "The allowed number of retries have been exceeded.";
-    this.missingOptimizedPaths = missingOptimizedPaths || [];
-    this.stack = (new Error()).stack;
+    var instance = new Error("The allowed number of retries have been exceeded.");
+
+    instance.name = "MaxRetryExceededError";
+    instance.missingOptimizedPaths = missingOptimizedPaths || [];
+
+    if (Object.setPrototypeOf) {
+        Object.setPrototypeOf(instance, Object.getPrototypeOf(this));
+    }
+
+    if (Error.captureStackTrace) {
+        Error.captureStackTrace(instance, MaxRetryExceededError);
+    }
+
+    return instance;
 }
 
-// instanceof will be an error, but stack will be correct because its defined
-// in the constructor.
-MaxRetryExceededError.prototype = new Error();
-MaxRetryExceededError.prototype.name = NAME;
+applyErrorPrototype(MaxRetryExceededError);
+
 MaxRetryExceededError.is = function(e) {
```

Comment:
delete the function and change `test/falcor/get/get.dataSource-only.spec.js` and `test/falcor/set/set.dataSource-only.spec.js` to use equality comparison with `MaxRetryExceededError.name` instead.